### PR TITLE
LOOP-3871: Make all alerts/notifications "Time Sensitive" 

### PR DESCRIPTION
**Pull request by rickpasetto**
https://tidepool.atlassian.net/browse/LOOP-3871
This is only the first step of many towards addressing the new iOS 15 "Time Sensitive" notifications.  It also contains a fix for the crasher https://tidepool.atlassian.net/browse/LOOP-3931

### CODE REVIEW

File: Loop/Managers/NotificationManager.swift [hunk 1]
```diff
@@ -98,6 +98,9 @@ extension NotificationManager {
 
         notification.body = body
         notification.sound = .default
+        if #available(iOS 15.0, *) {
+            notification.interruptionLevel = .timeSensitive
```
Comment (darinkrauss) [inline Loop/Managers/NotificationManager.swift:102]: [nit] Presumably we will add other notifications to be time sensitive, yes?

File: Loop/Managers/Alerts/UserNotificationAlertIssuer.swift [hunk 2]
```diff
@@ -68,7 +68,10 @@ fileprivate extension Alert {
         let userNotificationContent = UNMutableNotificationContent()
         userNotificationContent.title = content.title
         userNotificationContent.body = content.body
-        userNotificationContent.sound = getUserNotificationSound()
+        userNotificationContent.sound = userNotificationSound
+        if #available(iOS 15.0, *) {
+            userNotificationContent.interruptionLevel = backgroundContent?.isCritical == true ? .critical : .timeSensitive
```
Comment (darinkrauss) [inline Loop/Managers/Alerts/UserNotificationAlertIssuer.swift:73]: [nit] Or are all notifications time sensitive? I'm not sure every notification should be time sensitive. Perhaps I'm just misunderstanding this.

**Review by darinkrauss — APPROVED** [review]
A couple of questions, but LGTM otherwise.

File: Loop/Managers/Alerts/UserNotificationAlertIssuer.swift (hunk 2 above)
Comment (rickpasetto) [reply Loop/Managers/Alerts/UserNotificationAlertIssuer.swift:73]: @darinkrauss yes, for now.  There are more LOOP tickets on the way to address being more selective.

**Pull request merged**